python/sglang/srt/managers/tp_worker.py

```diff
@@ -141,12 +151,17 @@ def forward_batch_generation(
         self,
         model_worker_batch: ModelWorkerBatch,
         launch_done: Optional[threading.Event] = None,
+        skip_sample: bool = False,
     ):
         forward_batch = ForwardBatch.init_new(model_worker_batch, self.model_runner)
         logits_output = self.model_runner.forward(forward_batch)
         if launch_done:
             launch_done.set()
-        next_token_ids = self.model_runner.sample(logits_output, model_worker_batch)
+        if skip_sample:
+            next_token_ids = None
+        else:
+            next_token_ids = self.model_runner.sample(logits_output, model_worker_batch)
+        model_worker_batch.spec_info = forward_batch.spec_info
```

Comment:
This assignment should not be there.
The information flow should be `ScheduleBatch -> ModelWorkerBatch -> ForwardBatch`.
We will need this style to make all things work in overlap mode.